sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -128,6 +129,38 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudf-version-override=true
+      if (!conf.cudfVersionOverride) {
+        val props = new Properties
+        val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+        try {
+          props.load(cudfClassLoader.getResourceAsStream("cudf-java-version-info.properties"))
+        } catch {
+          case t: Throwable => {
```

Comment:
I don't see why we bother to catch this exception. `props` will be empty after this so `props.get("version")` will return `null` so we will get an NPE in just a few lines if this happens.

Reply:
I had added this for the case where  if we change the name of the property file. 
In this case we would get NPE here itself( and won't go to `props.get("version")` )  since `props.load` cannot load a valid file from InputStream.

Reply:
But like @jlowe said an NPE is not descriptive.  I would very much rather have an error that says that we could not find the properties file in the cudf jar so the versions do not match vs NPE or some error from props.load

Reply:
> I had added this for the case where if we change the name of the property file.

Rather than catching the error, I'd prefer to see the code check if the `getResourceAsStream` returned null and throw a more useful error mesasage.  Otherwise we'll get something a bit less than useful, like: `cudf properties file not found. NullPointerException`.  Plus it swallows the exception an allows the code to proceed which isn't desired.